### PR TITLE
refactor: Enhance the post forum endpoint after publishing a TP collection

### PR DESCRIPTION
**Pull request by kevinszuchet**

### CODE REVIEW

File: src/Forum/Forum.router.ts [hunk 1]
```diff
@@ -78,27 +75,13 @@ export class ForumRouter extends Router {
...
-          const { id: postId, link } = await createPost(
-            buildCollectionForumPost(collection, slicedItems)
-          )
-          await Collection.update<CollectionAttributes>(
-            { forum_link: link, forum_id: postId },
-            { id: collectionId }
-          )
-          return link
-        }
+        return this.service.upsertThirdPartyCollectionForumPost(
```
Comment (cyaiox) [inline src/Forum/Forum.router.ts:79]: should we wait for this upsert?

File: src/Collection/Collection.router.ts [hunk 2]
```diff
@@ -400,27 +395,10 @@ export class CollectionRouter extends Router {
...
-              result.collection,
-              result.items.slice(0, MAX_FORUM_ITEMS)
-            )
-          )
-          await Collection.update<CollectionAttributes>(
-            { forum_link: link, forum_id: postId },
-            { id }
-          )
-        }
+        this.forumService.upsertThirdPartyCollectionForumPost(
```
Comment (cyaiox) [inline src/Collection/Collection.router.ts:398]: should we wait for this upsert?

File: src/Forum/Forum.service.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    collection: ThirdPartyCollectionAttributes,
+    items: FullItem[]
+  ): Promise<string | undefined> {
+    if (collection.forum_id) {
+      const postData = await getPost(collection.forum_id)
+      await updatePost(
+        collection.forum_id,
+        buildCollectionForumUpdateReply(postData.raw, items)
+      )
+      return
```
Comment (cyaiox) [inline src/Forum/Forum.service.ts:24]: wdyt about returning the link here too?

File: src/Forum/Forum.router.ts (hunk 1 above)
Comment (kevinszuchet) [reply src/Forum/Forum.router.ts:79]: Yes, we should because of the catch. Thanks

File: src/Collection/Collection.router.ts (hunk 2 above)
Comment (kevinszuchet) [reply src/Collection/Collection.router.ts:398]: This was an error on my side. Thanks again!

File: src/Forum/Forum.service.ts (hunk 3 above)
Comment (kevinszuchet) [reply src/Forum/Forum.service.ts:24]: Is it possible to get it with the data that I have here or should I modify the `updatePost` so it returns the same as `createPost` and we can get the link there?

Comment (cyaiox) [reply src/Forum/Forum.service.ts]: I like how you updated the `updatePost` method to return the same link. sorry for the late reply

**Review by cyaiox — APPROVED** [review]
amazing job! 🚀

**Pull request merged**